pages/index.tsx

```diff
@@ -140,35 +179,27 @@ function SignedIn({ session, signOut, csrfToken }: SignedInProps) {
 
       <div className="flex flex-wrap mt-10">
 
-        {socialConnections.map(({ name, linked, connectUrl, description, connectedDescription, buttonLabel }) => (
-          <div key={name} className="mr-10 mt-10 flex flex-col items-center justify-between bg-gray-100 border rounded-sm p-5 w-[300px] h-[200px]">
-            <Image src={`/${name}.png`} alt={`${name} connection`} width={75} height={75} />
-            {linked ? (
-              <p>{connectedDescription}</p>) : (<>
-                <p>{description}</p>
-                <form action={connectUrl} method="post">
-                  <input type="hidden" name="csrfToken" value={csrfToken} />
-                  <input type="hidden" name="callbackUrl" value={window.location.origin} />
-                  <Button type="submit">{buttonLabel}</Button>
-                </form></>)}
-          </div>
+        {socialConnections.map(({ name, schema, linked, connectUrl, description, buttonLabel }) => (
+          <SocialAttestationProvider
+            key={name}
+            name={name}
+            schema={schema}
+            linked={linked}
+            connectUrl={connectUrl}
+            description={description}
+            buttonLabel={buttonLabel}
+            session={session}
+            attestMutation={attestMutation}
```

Comment:
When writing components which perform mutations, it is better that the components accepts a simple function instead of depending on an object returned by a library like `tanstack-query` (in this case you are passing the return value of `useMutation`).

Following up on the type suggestion I made above, you can define something like this:

```ts
interface AttestationProviderProps {
  attest: (schemaId: string) => void;
}
```

Then define props for `DiamondHandsAttestationProvider` and `SocialAttestationProvider` that extend `AttestationProviderProps`:

```ts
interface DiamondHandsAttestationProvider extends AttestationProviderProps {
// props specific to DiamondHands only
}

interface SocialAttestationProvider extends AttestationProviderProps {
 // props specific to SocialAttestation
}
}
```

Then you can pass to the component like this:

```ts
<SocialAttestationProvider
  attest={(schemaId) => attestMutation.mutate({ schemaId })
  ...
/>
```

This will make the components decoupled from the library we use to actually do the HTTP calls (and even agnostic to http, which makes it easier to unit test).

Reply:
This makes sense to me, nice being able to mock out the mutation function easily during testing. I have made the changes that _I think_ align with above priciple.